frontend/src/components/NewProject/Sync/index.jsx

```diff
@@ -76,46 +76,153 @@ const Sync = ({ setUserOp }) => {
     }
   }
 
+  const onDifferentName = async repoName => {
+    // Syncing can be instantaneous, avoid flickering by delaying setting the user operation.
+    const delayedSyncOp = setTimeout(() => setUserOp(SyncOp), 500)
+
+    setLoading(true)
+    setConflictModalOpen(false)
+    setMessage({
+      content: 'Processing the repository, this may take a while...',
+      color: 'green',
+    })
+
+    const repoURL = form.url
+
+    const res = await mirrorRepo(repoURL, uid, apiToken, repoName)
+    const migrateSuccessfully = res.ok
+
+    if (migrateSuccessfully) {
+      setMessage({
+        content: 'Migrated successfully, redirecting the project page...',
+        color: 'green',
+      })
+      await push(`/${username}/${repoName}`)
+    } else {
+      // If migration failed don't remove the uploading side.
+      clearTimeout(delayedSyncOp)
+      setUserOp(NoOp)
+
+      setLoading(false)
+      setMessage({
+        content: `Something went wrong. Are you sure "${form.url}" is a valid git repository?`,
+        color: 'red',
+      })
+    }
+  }
+
+  const onOverwrite = async () => {
+    // Syncing can be instantaneous, avoid flickering by delaying setting the user operation.
+    const delayedSyncOp = setTimeout(() => setUserOp(SyncOp), 500)
+
+    setLoading(true)
+    setConflictModalOpen(false)
+    setMessage({
+      content: 'Processing the repository, this may take a while...',
+      color: 'green',
+    })
+
+    const repoURL = form.url
+    const repoName = urlToName(repoURL)
+
+    const deletedRepoSuccessfully = await deleteRepo(
+      `${user.username}/${projectName}`,
+      apiToken,
+    )
```

Comment:
Deleting a repo should always be behind several confirmations. I feel like we shouldn't be offering the overwrite option on migration at all. The choice stems from the upload flow where we imagined a user might regularly upload a new file to update the project. 

Since we won't be letting users migrate repos themselves on alpha-2 I'd be happy to defer changing this to the next milestone though.

Reply:
#490